griptape/tasks/image_inpainting_task.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from typing import Callable
+
+from attr import define, field
+
+from griptape.artifacts import ImageArtifact, TextArtifact
+from griptape.tasks import BaseImageGenerationTask, BaseTask
+from griptape.utils import J2
+
+
+@define
+class ImageInpaintingTask(BaseImageGenerationTask):
+    """A task that modifies a select region within an image using a mask. Accepts a text prompt, image, and mask as
+    input in one of the following formats:
+    - tuple of (template string, ImageArtifact, ImageArtifact)
+    - tuple of (TextArtifact, ImageArtifact, ImageArtifact)
+    - Callable that returns a tuple of (TextArtifact, ImageArtifact, ImageArtifact)
+
+    Attributes:
+        image_generation_engine: The engine used to generate the image.
+        negative_rulesets: List of negatively-weighted rulesets applied to the text prompt, if supported by the driver.
+        negative_rules: List of negatively-weighted rules applied to the text prompt, if supported by the driver.
+        output_dir: If provided, the generated image will be written to disk in output_dir.
+        output_file: If provided, the generated image will be written to disk as output_file.
+    """
+
+    _input: tuple[str, ImageArtifact, ImageArtifact] | tuple[TextArtifact, ImageArtifact, ImageArtifact] | Callable[
+        [BaseTask], tuple[TextArtifact, ImageArtifact, ImageArtifact]
+    ] = field(default=None, kw_only=True)
+
+    @property
+    def input(self) -> (TextArtifact, ImageArtifact, ImageArtifact):
+        if (
+            isinstance(self._input, tuple)
+            and isinstance(self._input[0], TextArtifact)
+            and isinstance(self._input[1], ImageArtifact)
+            and isinstance(self._input[2], ImageArtifact)
+        ):
+            return self._input
+        if (
+            isinstance(self._input, tuple)
+            and isinstance(self._input[0], str)
+            and isinstance(self._input[1], ImageArtifact)
+            and isinstance(self._input[2], ImageArtifact)
+        ):
+            return (
+                TextArtifact(J2().render_from_string(self._input[0], **self.full_context)),
+                self._input[1],
+                self._input[2],
+            )
+        elif isinstance(self._input, Callable):
+            return self._input(self)
+        else:
+            raise ValueError("Input must be a tuple of (text, image, mask) or a callable that returns such a tuple.")
```

Comment:
I think this can be simplified to:

```python
    _input: tuple[TextArtifact | str, ImageArtifact, ImageArtifact] | Callable[
        [BaseTask], tuple[TextArtifact, ImageArtifact, ImageArtifact]
    ] = field(default=None, kw_only=True)

    @property
    def input(self) -> Tuple[TextArtifact, ImageArtifact, ImageArtifact]:
        if isinstance(self._input, tuple):
            if isinstance(self._input[0], str):
                input_text = TextArtifact(J2().render_from_string(self._input[0], **self.full_context))
            else:
                input_text = self._input[0]
            return (input_text, self._input[1], self._input[2])
        elif isinstance(self._input, Callable):
            return self._input(self)
        else:
            raise ValueError("Input must be a tuple of (text, image, mask) or a callable that returns such a tuple.")

```

Reply:
Do we lose the type checking on the input artifacts in indexes 1 + 2 in the simplified form or is that handled implicitly in the return type hint?